consensus/cryptarchia-engine/src/lib.rs

```diff
@@ -373,18 +373,32 @@ where
         }
     }
 
-    /// Create a new [`Cryptarchia`] instance with the updated state.
+    /// Process a block and returns a [`UpdatedCryptarchia`] that contains
+    /// an updated [`Cryptarchia`] instance
+    /// and the blocks pruned while processing the provided block.
     #[must_use = "Returns a new instance with the updated state, without modifying the original."]
-    pub fn receive_block(&self, id: Id, parent: Id, slot: Slot) -> Result<Self, Error<Id>> {
+    pub fn receive_block(
+        &self,
+        id: Id,
+        parent: Id,
+        slot: Slot,
+    ) -> Result<(Self, PrunedBlocks<Id>), Error<Id>> {
         let mut new: Self = self.clone();
         new.branches = new.branches.apply_header(id, parent, slot)?;
         new.local_chain = new.fork_choice();
-        new.update_lib();
-        Ok(new)
+        let pruned_blocks = new.update_lib();
+        Ok((new, pruned_blocks))
```

Comment:
Would not this be a single block always? :thinking:

Reply:
not necessarily, there could be three-headed (or more) forks in theory